xbmc/music/tags/MusicInfoTag.cpp

```diff
@@ -381,10 +381,10 @@ void CMusicInfoTag::SetAlbumArtist(const std::string& strAlbumArtist)
   }
 }
 
-void CMusicInfoTag::SetAlbumArtist(const std::vector<std::string>& albumArtists)
+void CMusicInfoTag::SetAlbumArtist(const std::vector<std::string>& albumArtists, bool FillDesc /* = false*/)
 {
   m_albumArtist = albumArtists;
-  if (m_strAlbumArtistDesc.empty()) 
+  if (m_strAlbumArtistDesc.empty() || FillDesc) 
```

Comment:
Never be sorry for questions, at least not until you have heard the answer!

It is all about the many uses that MusicInfoTag has. When we populate MusicInfoTag from scanned tags (for processing) we want to set both desc and vector from ARTIST and ALBUMARTIST tags. But when we populate MusicInfoTag from CSong and CAlbum having read them from the database or directly from a query (for display), sometimes we only have desc (no artist credits read from song_artist or album_artist and artist tables) and other times we can populate the artist vector too. But the ALBUMARTISTS and ARTISTS tags mean that desc is not always a concatenation of the vector, so we have to set then separately.

Initially it seemed simple to check if the desc was empty to confirm it had not been set separately (always setting it first it has a known value). But when APE tags are used to override ID3v2 we parse tiwce, and so desc could have been set but we want to override it with the APE value.